### PR TITLE
Remove scads of unused SASS, shared utility classes and update README

### PR DESCRIPTION
**Pull request by lyzadanger**
This PR removes a bunch of stuff and updates the README in the `styles` directory now that application components have been converted to tailwind.

The only utility styles coming from the shared package still in use are `.hyp-u-outline-on-keyboard-focus` and `.hyp-u-outline-on-keyboard-focus--inset`. I have some in-flight work on a branch for `frontend-shared` that reimplements this as a tailwind plugin. But for now, what I've opted to do is just add styles for these specific classes locally so that we can remove the other, unused utility classes from the output CSS. My locally-built (i.e. uncompressed) `sidebar.css` LOC reduces by about 20% from this change, so I think it's worth it! Buh-bye unused classes.

### CODE REVIEW

File: src/styles/sidebar/_utilities.scss [hunk 1]
```diff
@@ -1,3 +1,33 @@
...
+  box-shadow: 0 0 0 2px #59a7e8;
+}
+.js-focus-visible .u-outline-on-keyboard-focus:focus:not(.focus-visible) {
+  outline: none;
+  box-shadow: none;
+}
+.u-outline-on-keyboard-focus:focus:not(:focus-visible) {
+  outline: none;
+  box-shadow: none;
+}
```
Comment (robertknight) [inline src/styles/sidebar/_utilities.scss:19]: What is the reason for needing both of these selectors with the same rules? To override a default set elsewhere?

File: src/styles/sidebar/_utilities.scss [hunk 2]
```diff
@@ -1,3 +1,33 @@
-@use '@hypothesis/frontend-shared/styles/util';
-
 @use 'tailwindcss/utilities';
+
+// This is a temporary local implementation of `.hyp-u-outline-on-keyboard-focus`
+// and `.hyp-u-outline-on-keyboard-focus--inset`. These are the last remaining
+// utilities used by this application, so implementing them here allows us to
+// remove the `frontend-shared` utility styles from the `sidebar` CSS.
+// TODO: This should be re-implemented as a Tailwind plugin in `frontend-shared`
```
Comment (robertknight) [inline src/styles/sidebar/_utilities.scss:7]: What might the Tailwind version of this look like?

File: src/styles/README.md [hunk 3]
```diff
@@ -1,21 +1,6 @@
...
-- Tailwind is configured by this project's `tailwind.config.js`, which extends the [Tailwind preset provided by the `@hypothesis/frontend-shared` package](https://github.com/hypothesis/frontend-shared/blob/main/src/tailwind.preset.js) This configuration determines what utility classes are available. Add new needed Tailwind configuration properties as needed[^1] and consider extracting common properties to the `frontend-shared` package.
-- Check for unused SASS variables and mixins — and remove them.
-
-What we're aiming for at the end of this conversion process:
-
-- No SASS variables or mixins
-- External CSS minimal and within appropriate Tailwind layers
-
-[^1]: You will need to restart the local dev server after making configuration changes
+See https://github.com/hypothesis/frontend-toolkit/blob/master/docs/css-style-guide.md
```
Comment (robertknight) [inline src/styles/README.md:6]: I guess this document will need updating before too long to reflect the overall status of the BEM-ish => Tailwind project.

**Review by robertknight — APPROVED** [review]
LGTM. Couple of minor queries.

File: src/styles/sidebar/_utilities.scss (hunk 2 above)
Comment (lyzadanger) [reply src/styles/sidebar/_utilities.scss:7]: I could push a working branch to `frontend-shared` if you'd like to see.

File: src/styles/sidebar/_utilities.scss (hunk 1 above)
Comment (lyzadanger) [reply src/styles/sidebar/_utilities.scss:19]: This is a result of me simply c-n-p'ing the current output CSS for these utility classes. There may well be some redundancy here.

**Pull request merged**